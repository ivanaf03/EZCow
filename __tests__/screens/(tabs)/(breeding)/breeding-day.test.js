import React from "react";

import { render, waitFor } from "@testing-library/react-native";

import BreedingDay from "../../../../app/(tabs)/(breeding)/breeding-day";

describe("BreedingDay", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    const tree = render(
      <BreedingDay
        date="2022-01-01"
        events={[
          {
            id: "1",
            cowName: "Vaca Test",
            name: "Prueba",
            description: "Esto es una prueba",
          },
        ]}
      />
    );
    await waitFor(() => expect(tree).toMatchSnapshot());
  });

  it("should render correctly with no events", async () => {
    const tree = render(
      <BreedingDay
        date="2022-01-01"
        events={[
        ]}
      />
    );
    await waitFor(() => expect(tree).toMatchSnapshot());
  });
});