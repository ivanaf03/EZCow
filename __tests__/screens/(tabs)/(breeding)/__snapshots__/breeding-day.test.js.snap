// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedingDay should render correctly 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "backgroundColor": "#272727",
      },
      {
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
      },
      {
        "borderBottomColor": "#6bc0eb",
        "borderLeftColor": "#6bc0eb",
        "borderRightColor": "#6bc0eb",
        "borderTopColor": "#6bc0eb",
      },
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontWeight": "700",
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      2022-01-01
    </Text>
    <View
      style={
        [
          {
            "marginBottom": 8,
            "marginTop": 8,
          },
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontFamily": "Nunito-BoldItalic",
            },
            {
              "textDecorationLine": "underline",
            },
          ]
        }
      >
         - 
        Prueba
      </Text>
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "fontFamily": "Nunito-Medium",
            },
          ]
        }
      >
        Esto es una prueba
      </Text>
    </View>
  </View>
</View>
`;

exports[`BreedingDay should render correctly with no events 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "backgroundColor": "#272727",
      },
      {
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
      },
      {
        "borderBottomColor": "#6bc0eb",
        "borderLeftColor": "#6bc0eb",
        "borderRightColor": "#6bc0eb",
        "borderTopColor": "#6bc0eb",
      },
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontWeight": "700",
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      2022-01-01
    </Text>
  </View>
</View>
`;
